pkg/core/config/config.go

```diff
@@ -101,21 +101,12 @@ type Config struct {
 	Logging LogConfig `yaml:"logging" default:"{}"`
 	// Configuration of the managed collectd subprocess
 	Collectd CollectdConfig `yaml:"collectd" default:"{}"`
-	// If true, the agent will filter out [custom
-	// metrics](https://docs.signalfx.com/en/latest/admin-guide/usage.html#about-custom-bundled-and-high-resolution-metrics)
-	// without having to rely on the `whitelist.json` filter that was
-	// previously configured under `metricsToExclude`.  Whether a metric is
-	// custom or not is documented in each monitor's documentation.  If `true`,
-	// every monitor's default configuration (i.e. the minimum amount of
-	// configuration to make it work) will only send non-custom metrics.  In
-	// order to send out custom metrics from a monitor, certain config flags on
-	// the monitor must be set _or_ you can specify the metric in the
-	// `extraMetrics` config option on each monitor if you know the specific
-	// metric name.  You would not have to modify the whitelist via
-	// `metricsToInclude` as before.  If you set this option to `true`, the
-	// `whitelist.json` entry under `metricToExclude` should be removed, if it
-	// is present -- otherwise custom metrics won't be emitted.
-	EnableBuiltInFiltering bool `yaml:"enableBuiltInFiltering" default:"false"`
+	// This must be unset or explicitly set to true. In prior versions of the
+	// agent, there was a filtering mechanism that relied heavily on an
+	// external whitelist.json file to determine which metrics were sent by
+	// default.  This is all inherent to the agent now and the old style of
+	// filtering is no longer available.
+	EnableBuiltInFiltering *bool `yaml:"enableBuiltInFiltering" default:"true"`
```

Comment:
Don't think it technically needs to be a pointer since you're always setting a default value.

Reply:
It must be a pointer to make it default to `true` since the default value of bool is `false`.  Otherwise the default library doesn't know whether it was explicitly set to `false` or unset.